### ISSUE TITLE
Migration fail with "relation ... does not exist" error

### ISSUE THREAD

**Issue by lu-zen** [edgedb/edgedb#4465]
- EdgeDB Version: 2.1
- OS Version: endeavouros 22.4

Steps to Reproduce:

1. Apply provided migrations in order;
2. Migration 3 will fail;
Error:
```
Applied m1mamhqmoky3jfgoxvzb6kqwhrcfk647a4k6ty6gbe5tubozw6dewq (00001.edgeql)
Applied m1h6itucjtnrtsn5hu543lq44q5mzzv6zihcfxljrogfqhioyiczfq (00002.edgeql)
edgedb error: InternalServerError: relation "edgedbpub.4da643d4-405f-11ed-8ffa-094262b20fda" does not exist
  Hint: This is most likely a bug in EdgeDB. Please consider opening an issue ticket at https://github.com/edgedb/edgedb/issues/new?template=bug_report.md
  Server traceback:
      edb.errors.InternalServerError: relation "edgedbpub.4da643d4-405f-11ed-8ffa-094262b20fda" does not exist
```

[Migrations here
](https://github.com/lu-zen/edgedb-migration-error)

**Comment by lu-zen** [comment]
Tried to run migrations in a `2.3+ab48913` instance, the error persists.
```
Applied m1mamhqmoky3jfgoxvzb6kqwhrcfk647a4k6ty6gbe5tubozw6dewq (00001.edgeql)
Applied m1h6itucjtnrtsn5hu543lq44q5mzzv6zihcfxljrogfqhioyiczfq (00002.edgeql)
edgedb error: InternalServerError: relation "edgedbpub.266af6f9-4377-11ed-986f-43a8d5d8dd2e" does not exist
  Hint: This is most likely a bug in EdgeDB. Please consider opening an issue ticket at https://github.com/edgedb/edgedb/issues/new?template=bug_report.md
  Server traceback:
      edb.errors.InternalServerError: relation "edgedbpub.266af6f9-4377-11ed-986f-43a8d5d8dd2e" does not exist
```

**Comment by msullivan** [comment]
Should be fixed by #4513
